util/wsutil/wsutil.go

```diff
@@ -0,0 +1,40 @@
+package wsutil
+
+import (
+	"bytes"
+	"github.com/gorilla/websocket"
+	"sync"
+)
+
+type wrappedConnection interface {
+	WriteMessage(int, []byte) error
+}
+
+type WsWriterWrapper struct {
+	connection wrappedConnection
+
+	id  byte
+	buf bytes.Buffer
+	l   sync.Locker
+}
+
+func NewWsWriterWrapper(conn wrappedConnection, id byte, l sync.Locker) WsWriterWrapper {
```

Comment:
Can you keep `WsWriterWrapper` private and return an [`io.Writer`](https://golang.org/pkg/io/#Writer) here?